feat(STN-195): hero integration

silverli opened the pull request:
# READY FOR REVIEW

## Summary
Hero integration to allow it to be full-width on a basic page. The vertical margin of the full width hero at the top of the page will be different than a hero inserted with other components.

I also added a build task and refactored how hb-full-width works.

I tested and checked that default image styles (which is a responsive media field using hero image styles) are working as expected.

J and I paired on the mobile design, hero with no image, and hero with no overlay. Screenshots below:

#### No overlay
<img width="1440" alt="hero with no overla" src="https://user-images.githubusercontent.com/12848168/76031507-d3234280-5f05-11ea-92cc-3cf2a8a7f0fb.png">

#### No image
<img width="1440" alt="hero without image" src="https://user-images.githubusercontent.com/12848168/76031536-dc141400-5f05-11ea-835f-1490947252ec.png">
<img width="361" alt="Screen Shot 2020-03-05 at 5 16 48 PM" src="https://user-images.githubusercontent.com/12848168/76031546-dfa79b00-5f05-11ea-8140-6f63e2f037e0.png">


## Steps to Test
1. Add a hero to the home page using the hero image option. (See screenshot below). As for the display, go to Manage Display, and ensure that it's using the three column grid and the "Full width" section width option.
2. Do some browser testing and verify it looks good.
3. Go to another basic page, and add a hero to the page as the first thing using the paragraph type. Check out manage display for that page, and toggle back and forth between using the three column grid and the "Full width" section width option and the "limited width" option.
2. Do some browser testing and verify it looks good.

<img width="618" alt="Screen Shot 2020-03-05 at 12 56 49 PM" src="https://user-images.githubusercontent.com/12848168/76010619-2f27a000-5ee1-11ea-9feb-ebabc740b4b0.png">


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat(STN-195): first commit tpo get hero full width, also added a build task and refactored how hb-full-width works

Commit:
feat(STN-195): refining verticaal margin of the full width hero

Commit:
fix(STN-195): forgot a couple classes

Commit:
feat(STN-195): adds module updates for our purposes

Commit:
fix(STN-195): mobile style refineements and rename ui pattern field

heatherdesigns commented on docroot/themes/humsci/humsci_basic/package.json:
```diff
@@ -5,11 +5,12 @@
-    "watch:sass": "grunt watch",
     "build:sass": "grunt compile",
+    "build:js": "webpack",
+    "build": "npm run build:sass && npm run build:js",
```
:tada:

heatherdesigns commented on docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column.scss:
```diff
@@ -6,7 +6,11 @@
+  // option in layout builder, then apply our
+  // hb-page-width mixin. Otherwise, the section will
+  // go full width.
+  &:not(.hs-full-width) {
```
Excellent comments! :tada:

heatherdesigns reviewed (changes requested):
This is looking great! I still need to do browser testing but wanted to share some feedback thus far.

I found a couple things to look into.

If someone adds a link within the body of the hero, it displays as a button.
<img width="547" alt="style links in body" src="https://user-images.githubusercontent.com/12678977/76117177-55b90a00-5fb9-11ea-83d3-943daa9b16c5.png">

Lists are available in the WYSIWYG for the body but they do not display. Should we verify that is the expected behavior? Putting a list in the hero would be kinda weird. 🤷‍♀
![fyi lists do not display](https://user-images.githubusercontent.com/12678977/76117339-a2044a00-5fb9-11ea-9ed9-f8ce3544c175.png)

 The hero without an overly is an interesting scenario. This is a screenshot when logged out. This probably isn't notable when logged in because of the admin nav. There spacing in between the nav and hero image is awkward. And there's no space between the hero image and main content.

![no_overlay](https://user-images.githubusercontent.com/12678977/76117568-0f17df80-5fba-11ea-8de1-7ded63d802aa.png)

heatherdesigns commented:
Browser Testing:
- [x] Safari
- [x] Chrome
- [x] Chrome - Windows
- [x] Firefox - Mac
- [x] Firefox - Windows
- [x] Edge
- [x] IE

heatherdesigns reviewed (commented):
Browser testing is complete. Everything looks fantastic! Everything stacks in IE, which is perfect! It meets standards for IE. :tada:

The only thing I noticed in IE is that the image can get swishy at smaller screen sizes.

![IE](https://user-images.githubusercontent.com/12678977/76167955-fdad0f80-6140-11ea-9547-926510ba648d.png)

![another IE](https://user-images.githubusercontent.com/12678977/76167959-0998d180-6141-11ea-9f1e-9aa0d01f9d42.png)

---

One other thing to note. When I first opened the log in page, there's an extra space above the admin tool bar. I tried to debug it but couldn't figure it out. I'd refresh and nothing. So I opened the same page in another browser window and the extra space was gone. Super weird. I'm using Safari for my logged in state. I'm not sure if that has anything to do with it. I don't think we need to be too worried about it, but if we keep seeing it, then we should make a bug card for it.

![extra](https://user-images.githubusercontent.com/12678977/76168011-8f1c8180-6141-11ea-8821-5d55286de801.png)

Commit:
feat(STN-195): ensures links can render in hero without button styles

Commit:
fix(STN-195): fixes color contrast on links in hero and lists, as well as image alone without overlay and its margin

silverli commented:
@heatherdesigns in response to the things you noticed:

**List styles not showing up in the hero text:**
- the body is trimmed to 200 characters, so if you place lists towards the top, you'll see them :)
- also as for links showing up as buttons, I've fixed that now!
<img width="935" alt="Screen Shot 2020-03-09 at 10 27 30 AM" src="https://user-images.githubusercontent.com/12848168/76226395-32c66a00-61f4-11ea-9a7d-55bbcf635226.png">
<img width="461" alt="Screen Shot 2020-03-09 at 10 43 39 AM" src="https://user-images.githubusercontent.com/12848168/76226398-335f0080-61f4-11ea-9803-eedffa23cffe.png">
- I have also added better link and list styles for contrast based on Merani's guidance

**Hero image alone with extra vertical margin**
- I think I've fixed this with new styles that target when the image is alone

**Image gets squishy in IE11**
- I am not too worried. IE11 is a silver browser, and I think the functionality/appearance is good enough for that compliance. (Object-fit CSS property that we are using isn't supported in IE11... so that's why this is happening).

**Safari logged in state with weird bug**
- I wasn't able to replicate

Commit:
refactor(STN-195): couple last design edits
